device/transport/mqtt/test/_mqtt_twin_client_test.js

```diff
@@ -266,6 +266,32 @@ describe('MqttTwinClient', function () {
       assert.strictEqual(fakeMqttBase.publish.firstCall.args[1], JSON.stringify(fakePatch));
     });
 
+
+    /*Tests_SRS_NODE_DEVICE_MQTT_TWIN_CLIENT_41_001: [The `callback` shall be called with a Error object containing the error message if the `parsedMessage` contains an errorCode.]*/
+    it('calls its callback with an error when the response contains an errorcode', function (testCallback) {
+      var fakeTwin = {
+        fake: 'twin'
+      };
+
+      fakeMqttBase.publish = sinon.stub().callsFake(function (topic, body, options, callback) {
+        var requestId = topic.split('=')[1];
+        var fakeResponseTopic = '$iothub/twin/res/200?$rid=' + requestId;
+        callback();
+        fakeMqttBase.emit('message', fakeResponseTopic, body);
+      });
+
+      var twinClient = new MqttTwinClient(fakeMqttBase);
+      var fakeTwin = {
+        fake: 'patch',
+        errorCode: 'fakeErrorCode'
+      };
+      twinClient.updateTwinReportedProperties(fakeTwin, function (err, twin) {
+        assert.instanceOf(err, Error);
+        assert(!twin);
```

Comment:
>assert(!twin); [](start = 8, length = 14)

It has two other assert functions
assert.isNotOk
assert.isOk

Reply:
noted.